### PR TITLE
Compiling non-prod code only when explicitly specified

### PR DESCRIPTION
**Pull request by silva-fj**
Including dev only code when the `production` feature is not enabled makes it posible to include unsafe code to the production build in the cases where the feature has not been properly propagated. These changes switch the logic to include prod code by default and only include dev code when explicitly specified.

### CODE REVIEW

File: tee-worker/service/src/main_impl.rs [hunk 1]
```diff
@@ -98,7 +98,7 @@ pub(crate) fn main() {
 	// stdout from the service
 	#[cfg(feature = "production")]
```
Comment (silva-fj) [inline tee-worker/service/src/main_impl.rs]: Should we keep the `production` feature? or should we do:

```rust
 #[cfg(not(feature = "development"))]
 ```

the latter looks a bit weird but I think it makes more sense as `production` is the default now, I think having a production feature as well makes it a bit confusing

Comment (Kailai-Wang) [reply tee-worker/service/src/main_impl.rs]: I actually prefer latter as `production` should be the default now.

Is there a better name for `development`? :D 

cc @kziemianek too

Comment (arieljbirnbaum) [reply tee-worker/service/src/main_impl.rs]: `development` makes sense to me tbh :+1:

Comment (kziemianek) [reply tee-worker/service/src/main_impl.rs]: Agree, imho we should have only `development` feature as default code refers to production.

Comment (silva-fj) [reply tee-worker/service/src/main_impl.rs]: It makes sense, thank you guys! 👍🏼

File: bitacross-worker/enclave-runtime/src/rpc/worker_api_direct.rs [hunk 2]
```diff
@@ -49,8 +49,9 @@ use itp_top_pool_author::traits::AuthorApi;
 use itp_types::{DirectRequestStatus, RsaRequest, ShardIdentifier, H256};
 use itp_utils::{FromHexPrefixed, ToHexPrefixed};
 use jsonrpc_core::{serde_json::json, IoHandler, Params, Value};
+#[cfg(feature = "development")]
 use lc_scheduled_enclave::{ScheduledEnclaveUpdater, GLOBAL_SCHEDULED_ENCLAVE};
-use litentry_macros::if_not_production;
+use litentry_macros::if_development;
```
Comment (Kailai-Wang) [inline bitacross-worker/enclave-runtime/src/rpc/worker_api_direct.rs:54]: How about moving this line
```
use lc_scheduled_enclave::{ScheduledEnclaveUpdater, GLOBAL_SCHEDULED_ENCLAVE};
```

into the `if_development!({` block below?

There might be multiple places.

**Review by Kailai-Wang — APPROVED** [review]
Looks good in general, nicely done!

File: bitacross-worker/enclave-runtime/src/rpc/worker_api_direct.rs (hunk 2 above)
Comment (silva-fj) [reply bitacross-worker/enclave-runtime/src/rpc/worker_api_direct.rs:54]: I tried using the macro with `use` statements but it does not compile in some cases

e.g:
```
error: expected item after attributes
  --> app-libs/stf/src/trusted_call.rs:76:1
   |
76 | / if_development!({
77 | |     use crate::helpers::ensure_enclave_signer_or_alice;
78 | | });
   | |__^ in this macro invocation
   |
   = note: this error originates in the macro `if_development` (in Nightly builds, run with -Z macro-backtrace for more info)

```

**Pull request merged**